elide-integration-tests/src/test/java/com/yahoo/elide/tests/GraphQLIT.java

```diff
@@ -0,0 +1,702 @@
+                        )
+                ),
+                selection(
+                        field(
+                                "book",
+                                selections(
+                                        field("id", "0"),
+                                        field("title", "my book created in batch!")
+                                )
+                        )
+                )
+        ).toResponse();
+
+        compareJsonObject(
+                runQuery(toJsonArray(toJsonNode(graphQLRequest1), toJsonNode(graphQLRequest2))),
+                expectedResponse
+        );
+    }
+
+    @Test
+    public void runMultipleRequestsSameTransactionIncludingMutation() throws IOException {
+        Book book = new Book();
+        book.setId(1);
+        book.setTitle("1984");
+
+        Author author = new Author();
+        author.setId(1L);
+        author.setName("George Orwell");
+
+        String graphQLQuery = document(
+                mutation(
+                        selection(
+                                field(
+                                        "book",
+                                        arguments(
+                                                argument("op", "UPSERT"),
+                                                argument("data", book)
+                                        ),
+                                        selections(
+                                                field("id"),
+                                                field("title"),
+                                                field(
+                                                        "authors",
+                                                        arguments(
+                                                                argument("op", "UPSERT"),
+                                                                argument("data", author)
+                                                        ),
+                                                        selections(
+                                                                field("id"),
+                                                                field("name")
+                                                        )
+                                                )
+                                        )
+                                )
+                        )
+                )
+        ).toQuery();
+
+        create(graphQLQuery, null);
+
+        // This test demonstrates that multiple roots can be manipulated within a _single_ transaction
+        String graphQLRequest = document(
+                selections(
+                        field(
+                                "book",
+                                argument(
+                                        argument(
+                                                "ids",
+                                                Arrays.asList("1")
```

Comment:
This test is different than the deleted test.  The test has to both UPSERT and object in one root document, and fetch it back in another root document (all in the same query).